NewBank/newbank/server/NewBank.java

```diff
@@ -78,43 +81,78 @@ private String showMyAccounts(CustomerID customer) {
+
+        // convert second argument to double, return fail if not a number
+        double amount;
         try {
-            Double.parseDouble(splitRequest[1]);
+            amount = Double.parseDouble(splitRequest[1]);
         } catch (NumberFormatException e) {
+            // amount in invalid format - not a number
             return "FAIL";
         }
-        double amount = Double.parseDouble(splitRequest[1]);
-        //get the account objects (instead of the accountName)
-        ArrayList<Account> moveAccounts = new ArrayList<>();
-        for (int i = 0; i < customerAccounts.size(); i++) {
-            if (customerAccounts.get(i).getName().equals(splitRequest[2])) {
-                moveAccounts.add(customerAccounts.get(i));
-            }
+
+        // find the correct customer object
+        Customer customer = customers.get(customerId.getKey());
+
+        String fromAccountString = splitRequest[2];
+        String toAccountString = splitRequest[3];
+
+        // call move function in customer object
+        if (customer.move(fromAccountString, toAccountString, amount)) {
+            return "SUCCESS";
         }
-        for (int i = 0; i < customerAccounts.size(); i++) {
-            if (customerAccounts.get(i).getName().equals(splitRequest[3])) {
-                moveAccounts.add(customerAccounts.get(i));
-            }
+        return "FAIL";
+    }
+
+    private String payRequest(CustomerID customerId, String[] splitRequest) {
+        // check there are 2 things in the request
+        if (splitRequest.length != 3) {
+            return "FAIL";
         }
-        if (moveAccounts.size() != 2) { //check length of moveAccounts
+
+        // Current implementation only transfer money from Main to Main account
+        // In the future we need to allow user to specify from and to account name
+        String fromAccountName = "Main";
+        String toAccountName = "Main";
+
+        // 1. Get first customer
+        Customer fromCustomer = customers.get(customerId.getKey());
+        // 2. Find account of first customer
+        Account fromAccount = fromCustomer.getAccount(fromAccountName);
+
+        // 3. Find second customer
+        String toCustomerString = splitRequest[1];
+        if (!customers.containsKey(toCustomerString)) {
             return "FAIL";
         }
-        Account fromAccount = moveAccounts.get(0);
-        Account toAccount = moveAccounts.get(1);
-        if (customerAccounts.contains(fromAccount) && customerAccounts.contains(toAccount)) {
-            if (customers.get(customer.getKey()).move(fromAccount, toAccount, amount)) {
-                return "SUCCESS";
-            }
+        Customer toCustomer = customers.get(toCustomerString);
+
+        // 4. Find account of second customer
+        Account toAccount = toCustomer.getAccount(toAccountName);
+
+        // 5. Convert third argument to double, return fail if not a number
+        double amount;
+        try {
```

Comment:
Thanks for creating the pull request
I've found one bug - if I say 'PAY account -50' it will give me 50! Could you please add a check to make sure the double is positive? There may be a better way to do this :) Would be great if this could happen before the code is merged if possible 

<img width="426" alt="Screen Shot 2021-03-20 at 19 48 47" src="https://user-images.githubusercontent.com/80157394/111883867-c6a94500-89b5-11eb-814f-f41e74e6ac33.png">

Reply:
Great catch! We should also probably add this kind of check to addMoney as well, not only removeMoney. It will be done soon :)